pavelib/database.py

```diff
@@ -1,104 +1,126 @@
-    will remove any previously cached database files and apply migrations
-    on a fresh db. Additionally, the collective sha1 checksum for all of
-    these files will be written to file, for future comparisons/checking
-    for updates.
-
-    You can commit the resulting files in common/test/db_cache into
-    git to speed up test runs
+    Update and cache the MYSQL database for bokchoy testing:
+    * Remove any previously cached database files
+    * Apply migrations on a fresh db
+    * Write the collective sha1 checksum for all of these files to disk
+
+    WARNING: This will take several minutes.
     """
-    bokchoy_db_files = [
-        'bok_choy_data_default.json',
-        'bok_choy_data_student_module_history.json',
-        'bok_choy_migrations_data_default.sql',
-        'bok_choy_migrations_data_student_module_history.sql',
-        'bok_choy_schema_default.sql',
-        'bok_choy_schema_student_module_history.sql'
-    ]
     print('Removing cached db files for bokchoy tests')
-    for db_file in bokchoy_db_files:
-        try:
-            db_file_path = os.path.join(
-                '{}/common/test/db_cache'.format(Env.REPO_ROOT), db_file
-            )
-            os.remove(db_file_path)
-            print('\tRemoved {}'.format(db_file_path))
-        except OSError:
-            continue
-
-    sh('{}/scripts/reset-test-db.sh'.format(Env.REPO_ROOT))
-
-    # Write the fingerprint of the database files to disk for use in future
-    # comparisons
-    fingerprint = fingerprint_bokchoy_db_files()
-    with open('common/test/db_cache/bokchoy_migrations.sha1', 'w') as fingerprint_file:
-        fingerprint_file.write(fingerprint)
-
-
-def compare_bokchoy_db_fingerprints():
+    remove_files_from_folder(BOKCHOY_DB_FILES, CACHE_FOLDER)
+    apply_migrations(BOKCHOY_DB_FILES, update_cache_files=True)
+    compute_fingerprint_and_write_to_disk(MIGRATION_OUTPUT_FILES, ALL_DB_FILES)
+
+
+@needs('pavelib.prereqs.install_prereqs')
+@PassthroughTask
+@timed
+def update_local_bokchoy_db_from_s3():
+    """
+    Update the MYSQL database for bokchoy testing:
+    * Determine if your current cache files are up to date
+      with all the migrations
+    * If not then check if there is a copy up at s3
+    * If so then download then extract it
+    * Otherwise apply migrations as usual
+    """
+    fingerprint = fingerprint_bokchoy_db_files(MIGRATION_OUTPUT_FILES, ALL_DB_FILES)
+
+    if does_fingerprint_on_disk_match(fingerprint):
+        print ("DB cache files match the current migrations.")
+        apply_migrations(BOKCHOY_DB_FILES, update_cache_files=False)
+
+    elif is_fingerprint_in_bucket(fingerprint, CACHE_BUCKET_NAME):
+        print ("Found updated bokchoy db files at S3.")
+        refresh_bokchoy_db_cache_from_s3(fingerprint=fingerprint)
+        apply_migrations(BOKCHOY_DB_FILES, update_cache_files=False)
+        compute_fingerprint_and_write_to_disk(MIGRATION_OUTPUT_FILES, ALL_DB_FILES)
```

Comment:
Why are you computing the fingerprint here? These will be used on an ephemeral Jenkins box and thrown away. Do you intend on archiving the fingerprint file for debugging Jenkins builds?

Reply:
Because you have updated db cache files, the value for the fingerprint that is stored on disk will no longer match because it had been computed using your old ones.
Leaving those mismatched might break future assumptions. E.g. we might forget that use case when we move on to enabling this for devstack, where the user could run paver commands multiple times, and in any order.